### DISCUSSION TITLE
Running npm within /var/ subfolder

### DISCUSSION THREAD

**Discussion by DanielSiepmann** [netblue30/firejail#6405] [Q&A] [answered]
Hey, I'm pretty new and can't find a solution.

I have all my web dev projects within `/var/projects` sub folders and want to use `firejail` to make `npm` more secure.
But I can't find a way to use `npm` with `firejail` within the `/var/projects` subfolders. It always executes commands in my home folder.
I'd also like to lock it down to only write to the expected cache folders and the `/var/projects/` folder and sub folders.

Thanks in advance for help. Maybe I am just blind. I already tried to adjust via local files, using `--private-cwd`, added `writable-var` and a lot of other things.

**Comment by DanielSiepmann** [comment, marked as answer]
I could solve it on my own. I didn't get the loading right.
I modified my own `node.profile` which is not loaded by `npm.profile`.
I've adjusted both file to load a new `allow-var.local` file.

Is there any way to debug what's going on? The distributed files often contain the quiet directive which makes it hard to debug things.
What helped me was to use `firejail --profile=npm bash` to debug the environment and proof assertions I made.

**Reply by ghost** [reply]
Both `node.profile` and `npm.profile` redirect to `nodejs-common.profile`. If you place the content of `allow-var.local` in `nodejs-common.local` you could keep things organized in one override. Just ensure things don't get included twice. Also, note that none of the nodejs stack commands are in `firecfg` by default.

**Comment by ghost** [comment]
Can you provide some info, like ownership/permissions on `/var/projects`? Do you use a `global` npm configuration file like `/etc/npmrc` (pointing potential folders to /var/projects instead of your ${HOME})?

Using the below npm.local, what's the output of
`$ firejail --ignore=quiet /usr/bin/npm version`?

```console
$ cat ~/.config/firejail/npm.local
ignore noexec /var
private-cwd /var/projects
private-etc npmrc
writable-var
```
